network/phonebook.go

```diff
@@ -133,46 +223,33 @@ func (mp *MultiPhonebook) GetAddresses(n int) []string {
 
 	if len(mp.phonebookMap) == 1 {
 		for _, phonebook := range mp.phonebookMap {
-			return (*phonebook).GetAddresses(n)
+			return phonebook.GetAddresses(n)
 		}
 	}
 	sizes := make([]int, len(mp.phonebookMap))
 	total := 0
 	addrs := make([][]string, len(mp.phonebookMap))
-	names := make([]string, len(mp.phonebookMap))
-	i := 0
-	for name, p := range mp.phonebookMap {
-		names[i] = name
-		switch xp := (*p).(type) {
-		case *ArrayPhonebook:
-			sizes[i] = len(xp.Entries)
-		case *ThreadsafePhonebook:
-			sizes[i] = xp.Length()
-		default:
-			addrs[i] = xp.GetAddresses(1000)
-			sizes[i] = len(addrs[i])
-		}
-		total += sizes[i]
-		i++
-	}
-
-	addrSet := make(map[string]bool, total)
-	for pi, size := range sizes {
-		if addrs[pi] != nil {
-			mp.addAddressArrayToAdressSet(&addrSet, &(addrs[pi]))
-		} else {
-			xa := (*mp.phonebookMap[names[pi]]).GetAddresses(size)
-			mp.addAddressArrayToAdressSet(&addrSet, &xa)
+	uniqueEntries := make(map[string]bool, 0)
```

Comment:
simplification: since we're gathering everything into uniqueEntries, we could at the same time gather into one `addrs []string` and shuffle-select from that, and we don't need `sizes` anymore.
The original intent was not to ever gather a copy of all the addresses into one structure thus doubling the memory impact of all the addresses; but I guess that was probably premature optimization and all the addresses will probably never be over 1000 or 10000 and those are trivial numbers to deal with.

Reply:
sounds good; I'll make the update.